### PR TITLE
[WIP I'd removal]Chemical reagents/reactions code refactor to use ids associative lists

### PR DESCRIPTION
**Pull request by silicons**
subsystemizes some stuff too

### CODE REVIEW

File: code/modules/unit_tests/reagent_recipe_collisions.dm [hunk 1]
```diff
@@ -3,10 +3,10 @@
 /datum/unit_test/reagent_recipe_collisions
 
 /datum/unit_test/reagent_recipe_collisions/Run()
-	build_chemical_reactions_list()
+	reagents.Initialize()
```
Comment (Cyberboss) [inline code/modules/unit_tests/reagent_recipe_collisions.dm]: what are you doing kevin

File: code/modules/unit_tests/reagent_id_typos.dm [hunk 2]
```diff
@@ -3,12 +3,11 @@
 /datum/unit_test/reagent_id_typos
 
 /datum/unit_test/reagent_id_typos/Run()
-	build_chemical_reactions_list()
-	build_chemical_reagent_list()
+	SSreagents.Initialize()
```
Comment (Cyberboss) [inline code/modules/unit_tests/reagent_id_typos.dm]: Stop

**Comment by ExcessiveUseOfCobblestone** [comment]
I hate the ID system boo boo!

**Comment by Cyberboss** [comment]
RE 👏 MOVE 👏 STRING 👏 ID 👏 SYSTEMS

**Comment by SpaceManiac** [comment]
making it associative is a good step 1 to removing the string IDs so I approve

**Comment by Razharas** [comment]
>Trying to fix chem system IDs by making them clean and subsystimized
But why?

**Comment by kevinz000** [comment]
>1000 compile errors

**Pull request closed without merging**